modules/nextflow/src/main/groovy/nextflow/processor/PublishDir.groovy

```diff
@@ -508,9 +508,7 @@ class PublishDir {
             makeDirs(path)
         }
         catch( Throwable e ) {
-            log.warn "Failed to create publish directory: ${path.toUriString()} -- See log file for details", e
-            if( NF.strictMode || failOnError )
-                session?.abort(e)
+            session?.abort(new IllegalStateException("Failed to create publish directory: ${path.toUriString()}", e))
```

Comment:
I think it's enough `throw new Illegal..`. There already many usage like that in this class

Reply:
If I just re-throw the exception, the resulting error is:
```
ERROR ~ Error executing process > 'foo'

Caused by:
  The provided token is malformed or otherwise invalid ...

 -- Check '.nextflow.log' file for details
```

Ideally it would show the "failed to create publish directory: <path>" in the console output, but instead it is only in the log.

This is still an improvement though, since before it didn't say anything about the publish directory. Maybe the console output is a broader issue with publish errors. I would be fine with applying your suggestion and merging for now.

Reply:
```suggestion
            throw new IllegalStateException("Failed to create publish directory: ${path.toUriString()}", e)
```